SUSE15 sp2 changes

Himanshu-xilinx opened the pull request:
1. Added SUSE cmake packaging support
2. Updated xrtdeps.sh script for SUSE dependencies
3. Updated pkg_hw_platform.sh script for SUSE.

gbuildx commented:
Build Passed!

chvamshi-xilinx reviewed (approved):
Looks good.

stsoe commented on src/runtime_src/core/tools/xbmgmt2/CMakeLists.txt:
```diff
@@ -1,4 +1,15 @@
 # -----------------------------------------------------------------------------
+if(NOT WIN32)
+if ((${LINUX_FLAVOR} STREQUAL SUSE))
+set(Boost_USE_STATIC_LIBS OFF)               # find dynamic libraries
+else()
+set(Boost_USE_STATIC_LIBS ON)               # Only find static libraries
+endif()
+else()
+set(Boost_USE_STATIC_LIBS ON)               # Only find static libraries
+endif()
+set(Boost_USE_MULTITHREADED ON)             # Multi-threaded libraries
+
```
Boost configuration must no be controlled in CMakeLists.txt post 2020.2.  If you notice this has changed in master as compared to 2020.2, we link dynamically with Boost for everything now.  Please remove.

Himanshu-xilinx replied:
done

stsoe commented on src/runtime_src/core/tools/xbutil2/CMakeLists.txt:
```diff
@@ -1,4 +1,15 @@
 # -----------------------------------------------------------------------------
+if(NOT WIN32)
+if ((${LINUX_FLAVOR} STREQUAL SUSE))
+set(Boost_USE_STATIC_LIBS OFF)               # find dynamic libraries
+else()
+set(Boost_USE_STATIC_LIBS ON)               # Only find static libraries
+endif()
+else()
+set(Boost_USE_STATIC_LIBS ON)               # Only find static libraries
+endif()
+set(Boost_USE_MULTITHREADED ON)             # Multi-threaded libraries
+
```
See comment on xbmgmt2

Himanshu-xilinx replied:
done

stsoe reviewed (changes requested):
Just two comments regarding Boost.   Otherwise looks good.

rozumx reviewed (changes requested):
There are a lot of important changes in the pull request and yet the description of what was done is empty.   **_At least_** one would expect something like:
1. Added SUSE cmake packaging support
2. Updated xrtdeps.sh script for SUSE dependencies
3.  Updated xbutil2 & xbmgmt2 to use boost dynamic libraries for SUSE builds
4. Updated "classic" packager to be SUSE aware.

In addition, there are some unexpected changes to ulite.c, which "seems" out-of-place.

gbuildx commented:
Build Failed! :(

xilinxgitops commented:
retest this please

gbuildx commented:
Build Failed! :(

xilinxgitops commented:
retest this please

gbuildx commented:
Build Failed! :(

xilinxgitops commented:
retest this please

gbuildx commented:
Build Passed!

Himanshu-xilinx commented:
> There are a lot of important changes in the pull request and yet the description of what was done is empty. **_At least_** one would expect something like:
> 
> 1. Added SUSE cmake packaging support
> 2. Updated xrtdeps.sh script for SUSE dependencies
> 3. Updated xbutil2 & xbmgmt2 to use boost dynamic libraries for SUSE builds
> 4. Updated "classic" packager to be SUSE aware.
> 
> In addition, there are some unexpected changes to ulite.c, which "seems" out-of-place.

added comments @rozumx review

rozumx reviewed (approved):
Approved
